### PR TITLE
ServiceNowITAdmin:Added new integration

### PR DESCRIPTION
**Pull request by shjain-799**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
NA: It's a new Integration

## Description
Initial commit for integration.

## Screenshots
NA

## Minimum version of Demisto
- [x] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation 

## Demisto Partner?
- [ ] The title must be in the following format: **[YOUR_PARTNER_ID] short description**
PANW IT team

### CODE REVIEW

File: Packs/ServiceNowITAdmin/Integrations/ServiceNow_IT_Admin/ServiceNow_IT_Admin.py [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+    """
+        PARSE AND VALIDATE INTEGRATION PARAMS
+    """
+    params = demisto.params()
+    api_version = params.get('api_version', None)
+
+    # get the service API url
+    base_url = urljoin(params.get('url').strip('/'), BASE_URI)
+    if api_version:
+        base_url += api_version
```
Comment (dantavori) [inline Packs/ServiceNowITAdmin/Integrations/ServiceNow_IT_Admin/ServiceNow_IT_Admin.py:475]: why do we let the client choose the api version and not define it ourselves?

File: Packs/ServiceNowITAdmin/Integrations/ServiceNow_IT_Admin/ServiceNow_IT_Admin.py [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+        proxy=proxy)
+
+    try:
+        if command in commands:
+            human_readable, outputs, raw_response = commands[command](client, demisto.args())
+            return_outputs(readable_output=human_readable, outputs=outputs, raw_response=raw_response)
+    # Log exceptions
+    except Exception:
+        demisto.error(f'Failed to execute {demisto.command()} command. Traceback: {traceback.format_exc()}')
+        return_error(f'Failed to execute {demisto.command()} command. Traceback: {traceback.format_exc()}')
```
Comment (dantavori) [inline Packs/ServiceNowITAdmin/Integrations/ServiceNow_IT_Admin/ServiceNow_IT_Admin.py:511]: just the return_error is enough here

File: Packs/ServiceNowITAdmin/Integrations/ServiceNow_IT_Admin/ServiceNow_IT_Admin.py [hunk 3]
```diff
@@ -0,0 +1,515 @@
...
+
+        Returns:
+            success : success=True, id, email, login as username, details, active status
+            fail : success=False, id, login as username, errorCod, errorMessage, details
+    """
+    scim = verify_and_load_scim_data(args.get('scim'))
+    parsed_scim_data = map_scim(scim)
+    user_id = parsed_scim_data.get('id')
+
+    if not (user_id):
```
Comment (dantavori) [inline Packs/ServiceNowITAdmin/Integrations/ServiceNow_IT_Admin/ServiceNow_IT_Admin.py:375]:
```suggestion
    if not user_id:
```

File: Packs/ServiceNowITAdmin/Integrations/ServiceNow_IT_Admin/ServiceNow_IT_Admin.py [hunk 4]
```diff
@@ -0,0 +1,515 @@
...
+            fail : success=False, id, login as username, errorCod, errorMessage, details
+    """
+    old_scim = verify_and_load_scim_data(args.get('oldScim'))
+    new_scim = verify_and_load_scim_data(args.get('newScim'))
+    custom_mapping = demisto.params().get('customMappingUpdateUser')
+
+    parsed_old_scim = map_scim(old_scim)
+    user_id = parsed_old_scim.get('id')
+
+    if not (user_id):
```
Comment (dantavori) [inline Packs/ServiceNowITAdmin/Integrations/ServiceNow_IT_Admin/ServiceNow_IT_Admin.py:324]:
```suggestion
    if not user_id:
```

File: Packs/ServiceNowITAdmin/Integrations/ServiceNow_IT_Admin/ServiceNow_IT_Admin.py [hunk 5]
```diff
@@ -0,0 +1,515 @@
...
+
+        Returns:
+            success : success=True, id, email, login as username, details, active status
+            fail : success=False, id, login as username, errorCod, errorMessage, details
+    """
+    scim = verify_and_load_scim_data(args.get('scim'))
+    parsed_scim_data = map_scim(scim)
+    user_id = parsed_scim_data.get('id')
+
+    if not (user_id):
```
Comment (dantavori) [inline Packs/ServiceNowITAdmin/Integrations/ServiceNow_IT_Admin/ServiceNow_IT_Admin.py:426]:
```suggestion
    if not user_id:
```

File: Packs/ServiceNowITAdmin/Integrations/ServiceNow_IT_Admin/ServiceNow_IT_Admin.py (hunk 1 above)
Comment (sonaaru) [reply Packs/ServiceNowITAdmin/Integrations/ServiceNow_IT_Admin/ServiceNow_IT_Admin.py:475]: @dantavori It's an optional parameter, if the user wants to run the particular version of API they needs to mention it here else latest will be called.

File: Packs/ServiceNowITAdmin/Integrations/ServiceNow_IT_Admin/ServiceNow_IT_Admin.py (hunk 2 above)
Comment (sonaaru) [reply Packs/ServiceNowITAdmin/Integrations/ServiceNow_IT_Admin/ServiceNow_IT_Admin.py]: removed demisto.error() line

File: Packs/ServiceNowITAdmin/Integrations/ServiceNow_IT_Admin/ServiceNow_IT_Admin.py (hunk 3 above)
Comment (sonaaru) [reply Packs/ServiceNowITAdmin/Integrations/ServiceNow_IT_Admin/ServiceNow_IT_Admin.py]: Changed the code

File: Packs/ServiceNowITAdmin/Integrations/ServiceNow_IT_Admin/ServiceNow_IT_Admin.py (hunk 4 above)
Comment (sonaaru) [reply Packs/ServiceNowITAdmin/Integrations/ServiceNow_IT_Admin/ServiceNow_IT_Admin.py]: Changed the code

File: Packs/ServiceNowITAdmin/Integrations/ServiceNow_IT_Admin/ServiceNow_IT_Admin.py (hunk 5 above)
Comment (sonaaru) [reply Packs/ServiceNowITAdmin/Integrations/ServiceNow_IT_Admin/ServiceNow_IT_Admin.py]: Changed the code

**Review by dantavori — CHANGES_REQUESTED** [review]

File: Packs/ServiceNowITAdmin/TestPlaybooks/playbook-ServiceNowITAdmin-Test.yml [hunk 6]
```diff
@@ -0,0 +1,609 @@
+id: playbook-ServiceNowITAdmin-Test
+name: playbook-ServiceNowITAdmin-Test
+version: 5
```
[the remark this answers is not in the document]
Comment (sonaaru) [reply Packs/ServiceNowITAdmin/TestPlaybooks/playbook-ServiceNowITAdmin-Test.yml]: Done the chnages

File: Packs/ServiceNowITAdmin/TestPlaybooks/playbook-ServiceNowITAdmin-Test.yml [hunk 7]
```diff
@@ -0,0 +1,609 @@
+id: playbook-ServiceNowITAdmin-Test
```
[the remark this answers is not in the document]
Comment (sonaaru) [reply Packs/ServiceNowITAdmin/TestPlaybooks/playbook-ServiceNowITAdmin-Test.yml]: Moved the play book inside NonCircleTests dir

**Review by dantavori — APPROVED** [review]

**Pull request merged**